### ISSUE TITLE
Help with review of characterization of physics task types to report to CMS monitoring

### ISSUE THREAD

**Issue by khurtado** [cms-sw/cmssw#42587]
We would like to review and update/improve the current monitoring system that reports the physics activity per job for production, current procedure and physics types were created years ago and types like NANOAOD are not included for example.

* Overview of the current monitoring:
We currently support the following physics/task types at the monitoring level [[source](https://github.com/dmwm/cms-htcondor-es/blob/master/src/htcondor_es/convert_to_json.py#L1001-L1045)]:

- GENSIM
- DIGI
- RECO
- DIGIRECO
- MINIAOD

There is no combination for the physics types above. Which means we won't see a job with: GENSIM+DIGI+RECO at the monit level at present for example, even if we have such a job in the system. Tagging GEN-only jobs is not currently considered either. As far as we know, this list was created many years ago (~5) and does not include e.g.: NANOAOD either. 

This logic is based on task names defined in requests (in ReqMgr2) but we have noticed many mis-identifications in the jobs, which is why we want to use a different source for the logic (e.g.: the origin of this source, which would be cmsDriver and is a feature request: [link here](https://github.com/cms-sw/cmssw/issues/42450))

Independently of the Github issue above, we would like feedback on the relevant physics types that should actually be supported in the CMS job monitoring.

For instance, NANOAOD should be included, but we also think the granularity of the physics types per job is not accurate at present. E.g.: A stepChain workflow in production can run:

 **GEN,SIM,DIGI,RECO,MINIAOD,NANOAOD**

everything in the same job. Hence, all those physics types should be included. For consistency, that means we would get rid of:

- GENSIM
- DIGIRECO

But start including, e.g.:
**- GEN
- SIM
- DIGI
- RECO
- MINIAOD
- NANOAOD**

and a combination of the types above with commas for example: **GEN,SIM,DIGI,RECO**, etc.

So, here are some question we would appreciate some feedback:

- Would the above be reasonable, or would you propose a different way to show this in monitoring?
- Does the list above (GEN,SIM,DIGI,RECO,MINIAOD,NANOAOD) containts all the relevant physics/task types, or should we be including some additional ones?

Tagging @amaltaro

**Comment by makortel** [comment]
assign pdmv

**Comment by sunilUIET** [comment]
Hi @khurtado 

   The proposed task type looks ok to me. Do you want to add specific tags for wmLHE, PU/noPU samples as well or there is sufficient information available to tag these processes?

**Comment by khurtado** [comment]
@sunilUIET Yes please! Also, the step names to physics type mapping we have so far are below, could you let us know if these are okay as well or if we are missing anything?:

```
step argument -> physics type name
GEN -> GEN
SIM -> SIM
DIGI -> DIGI
RECO -> RECO
PAT -> MINIAOD
NANO -> NANOAOD

step:XXX = step. E.g.: DIGI:pdigi_valid = DIGI
```

**Comment by makortel** [comment]
Just thinking out loud

> ```
> DIGI -> DIGI
> ```

would it be useful for this monitoring use case to separate the three meanings of "DIGI"
* DIGI without pileup
* DIGI with premixing (i.e. overlaying premixed pileup events)
* DIGI with classical mixing (e.g. creation of premixed pileup library, samples for which premixing is not used

?

I'm then wondering if this monitoring is used to in a way where the (expected) resource needs play a role? If yes, should e.g. DQM be included? (DQM increases the application memory need, and is apparently nowadays part of the data reprocessing)

**Comment by sunilUIET** [comment]
> @sunilUIET Yes please! Also, the step names to physics type mapping we have so far are below, could you let us know if these are okay as well or if we are missing anything?:
> 
> ```
> step argument -> physics type name
> GEN -> GEN
> SIM -> SIM
> DIGI -> DIGI
> RECO -> RECO
> PAT -> MINIAOD
> NANO -> NANOAOD
> 
> step:XXX = step. E.g.: DIGI:pdigi_valid = DIGI
> ```

Hi @khurtado 

  This looks reasonable to me.

**Comment by khurtado** [comment]
@makortel  If these can be identified from the `--step` arguments, then I see no harm in implementing them in monit.

**Comment by makortel** [comment]
> If these can be identified from the `--step` arguments, then I see no harm in implementing them in monit. Do you have a pattern recipe for this?

(using the nomenclature of https://github.com/cms-sw/cmssw/issues/42587#issuecomment-1684465286)

The "DIGI with premixing" has both `DIGI` and `DATAMIX` in the `--step` arguments (as well as `--datamix PreMix` argument, and the `--pileup_input` argument mentioned below).

The "DIGI without pileup" and "DIGI with classical mixing" have `DIGI` in the `--step` arguments.

As far as I know, the only way differentiate between "DIGI without pileup" and "DIGI with classical mixing" at the `cmsDriver.py` command line argument level are the presence of `--pileup` and `--pileup_input` arguments. If they are present, the job is "DIGI with classical mixing", if they are not, the job is "DIGI without pileup".

**Comment by khurtado** [comment]
@makortel Thank you!
